voyager-hilt/src/main/java/cafe/adriel/voyager/hilt/ViewModel.kt

```diff
@@ -2,101 +2,138 @@ package cafe.adriel.voyager.hilt
-    val viewModelStoreOwner = checkNotNull(LocalViewModelStoreOwner.current) {
+public inline fun <reified T : ViewModel> Screen.hiltViewModel(
+    key: String? = null,
+    viewModelProviderFactory: ViewModelProvider.Factory? = null,
+    viewModelStoreOwner: ViewModelStoreOwner = checkNotNull(LocalViewModelStoreOwner.current) {
         "No ViewModelStoreOwner was provided via LocalViewModelStoreOwner"
+    },
+    extras: CreationExtras = if (viewModelStoreOwner is HasDefaultViewModelProviderFactory) {
+        viewModelStoreOwner.defaultViewModelCreationExtras
+    } else {
+        CreationExtras.Empty
     }
-    return remember(key1 = T::class) {
-        val hasDefaultViewModelProviderFactory = requireNotNull(lifecycleOwner as? HasDefaultViewModelProviderFactory) {
-            "$lifecycleOwner is not a androidx.lifecycle.HasDefaultViewModelProviderFactory"
-        }
-        val viewModelStore = requireNotNull(viewModelStoreOwner?.viewModelStore) {
-            "$viewModelStoreOwner is null or have a null viewModelStore"
-        }
-        val factory = VoyagerHiltViewModelFactories.getVoyagerFactory(
-            activity = context.componentActivity,
-            delegateFactory = viewModelProviderFactory
-                ?: hasDefaultViewModelProviderFactory.defaultViewModelProviderFactory
-        )
-        val provider = ViewModelProvider(
-            store = viewModelStore,
-            factory = factory,
-            defaultCreationExtras = hasDefaultViewModelProviderFactory.defaultViewModelCreationExtras
-        )
-        provider[T::class.java]
+): T = getViewModel<T, Nothing>(
+    key = key,
+    viewModelProviderFactory = viewModelProviderFactory,
+    viewModelStoreOwner = viewModelStoreOwner,
+    extras = extras
+)
+
+/**
+ * A function to provide a [dagger.hilt.android.lifecycle.HiltViewModel] managed by voyager [Screen] lifecycle
+ * instead of using Activity or Fragment LifecycleOwner.
+ *
+ * @param viewModelFactory A custom callback to be used for assisted injection
+ * @param key The key used to identify the [ViewModel]. Default is null
+ * @param viewModelStoreOwner The owner of the [ViewModel] that controls the scope and lifetime
+ * of the returned [ViewModel]. Defaults to using [LocalViewModelStoreOwner].
+ * @param extras The default extras used to create the [ViewModel].
+ *
+ * @return A [ViewModel] that is an instance of the given [T] type.
+ */
+@Composable
+public inline fun <reified T : ViewModel, VMF> Screen.hiltViewModel(
+    noinline viewModelFactory: ViewModelFactory<VMF>,
+    key: String? = null,
+    viewModelStoreOwner: ViewModelStoreOwner = checkNotNull(LocalViewModelStoreOwner.current) {
+        "No ViewModelStoreOwner was provided via LocalViewModelStoreOwner"
+    },
+    extras: CreationExtras = if (viewModelStoreOwner is HasDefaultViewModelProviderFactory) {
+        viewModelStoreOwner.defaultViewModelCreationExtras
+    } else {
+        CreationExtras.Empty
     }
```

Comment:
Perhaps have `viewModelFactory` as the last parameter to avoid needing parentheses or named parameters

```suggestion
    key: String? = null,
    viewModelStoreOwner: ViewModelStoreOwner = checkNotNull(LocalViewModelStoreOwner.current) {
        "No ViewModelStoreOwner was provided via LocalViewModelStoreOwner"
    },
    extras: CreationExtras = if (viewModelStoreOwner is HasDefaultViewModelProviderFactory) {
        viewModelStoreOwner.defaultViewModelCreationExtras
    } else {
        CreationExtras.Empty
    },
    noinline viewModelFactory: ViewModelFactory<VMF>
```